app/routers/api/files.py

```
@@ -128,7 +128,137 @@ def _serialize_file(
-@router.get("")
+def _parse_tag_ids(tag_ids: Optional[str]) -> Optional[List[int]]:
+    """Parse comma-separated tag IDs string into a list of integers."""
+    if not tag_ids:
+        return None
+    try:
+        return [int(tid.strip()) for tid in tag_ids.split(",") if tid.strip()]
+    except ValueError:
+        raise ValueError("Invalid tag_ids format. Must be comma-separated integers.")
+
+
+def _apply_filters(
```

Ruff (0.15.9)

[warning] Too many branches (18 > 12)

(PLR0912)